### DISCUSSION TITLE
Chrome Android Issues

### DISCUSSION THREAD

**Discussion by jamieboy2020** [vpnhood/VpnHood#148] [General]
First of all just want to say thanks for your work on this project, works very well. There is just one issue I have noticed when using the VpnHood client on Android. I have installed a server on my home machine, both on Linux and Windows. I have now installed the server on a cloud hosted Linux server. I can connect to all of them fine using the Android VpnHood App. But when using Google Chrome, certain websites do not work and give the attached error. In Firefox all sites seem to work fine. In Edge no sites will work at all. Does anyone else experience this in Chrome / Edge on Android?

![Screenshot_20221026-222757_Chrome](https://user-images.githubusercontent.com/61209945/198142246-8b2c9eea-9eb5-4437-80c9-50a833da19f6.jpg)

**Comment by trudyhood** [comment]
Are you using UDP mode or TCP in VpnHood Client?

**Comment by jamieboy2020** [comment]
Hi Trudy. Both UDP and TCP give the same error on certain pages. I tried UDP just now and noticed a different error on the Google homepage. Seems like the error is related to the UDP protocol, not sure if it gives any clues as to why the error may occur in Chrome. All sites also load fine in UDP in Firefox.
![Screenshot_20221026-231657_Chrome](https://user-images.githubusercontent.com/61209945/198149475-886d9e37-404b-43e2-84a2-7b87f643a4e9.jpg)

**Comment by trudyhood** [comment]
Perhaps there is some issue in DNS packets; maybe one firefox uses DNS secure. I wish we could simulate the problem. Can you send us a diagnose and give me the diagnose reference ID?

**Reply by jamieboy2020** [reply]
Just ran a diagnosis. a97e0e03@20221026T230021-06aab27d

**Comment by trudyhood** [comment]
I got it. 
1) Is this issue exists in android only?
2) Can you try with another android too?
3) Are you using DNS secure? (https://www.howtogeek.com/795644/how-to-enable-secure-private-dns-on-android/)

**Comment by jamieboy2020** [comment]
Hi Trudy. Sorry it took me a while to get a chance to test on another device. I installed the VPNHood app on my windows laptop and I receive the same problem on some websites in Chrome and Edge. Firefox works fine for all sites. Also I tried switching off VPN secure on my phone and problem still exists. I have also tried using the data connection on my phone and issue stil persists.I tried installing the app on another Android phone and the same problem on there too. 

The public server does not have this issue. Do you know if the public server is using the same Server version as the one I am installing on all my machines from the script?https://github.com/vpnhood/VpnHood/wiki/How-to-run-VpnHood-Server-on-Linux

**Comment by trudyhood** [comment]
We use the same server version. I think  the problem may be something DNS. Can you send us a server AccessKey to support@vpnhood.com

**Comment by bazgard** [comment]
I have the same problem too,
This problem exists in Iran on blocked sites such as Facebook, Twitter, etc. when using Google Chrome
Also, changing DNS from Google Chrome settings did not solve the problem
This problem does not exist on Firefox Android browser

**Reply by trudyhood** [reply]
It sounds like it is a severe issue. Unfortunately, we couldn't simulate the problem. Is there any volunteer to give us AnyDesk for a few days? So we can keep working on it at that Network? send us an email at support@vpnhood.com

**Comment by minamaged-dev** [comment]
Had the same issue when hosting the server on aws
Solution was to assign an ipv6 to the server machine

**Reply by trudyhood** [reply]
does it solve the issue?

**Reply by minamaged-dev** [reply]
yes, it resolved the issue.
I might make a cloudformation template when I have free time to automate the whole process.

**Comment by AliMoal** [comment]
I have the same problem on Android and even Windows. I was wondering If clients could set DNS settings, It might be helpful.

**Reply by trudyhood** [reply]
What do you mean by "clients set DNS settings"? Please give me an example.

**Comment by AliMoal** [comment]
I mean in client app, there could be a section to add our own DNS settings. such as Cloudflare (1.1.1.1) or Google (8.8.8.8). Because I think this problem can be solved by changing DNS.

**Reply by trudyhood** [reply]
VpnHood always changes DNS to 8.8.8.8
Perhaps the issue exists somewhere else. Can you send me the diagnosis log file? Maybe I can find the issue.

**Reply by AliMoal** [reply]
Ok, I'll send you the file as soon as possible, But in my opinion It's better to make DNS settings changeable.

**Reply by trudyhood** [reply]
>I'll send you the file as soon as possible

Thank you

> in my opinion It's better to make DNS settings changeable.

I am interested to know the benefits of this idea.

**Reply by AliMoal** [reply]
Some DNS servers do not work perfectly (even Google) In Iran. It's good for user to be free for choosing the best DNS server.

**Reply by trudyhood** [reply]
> Some DNS servers do not work perfectly (even Google) In Iran. It's good for user to be free for choosing the best DNS server.

VpnHood changes DNS to 8.8.8.8 and tunnels the DNS request through VPN Tunnel, So it doesn't matter what you set for DNS because it will go through VPN tunnel. This issue exists somewhere else and can not be solved by changing the DNS.

**Reply by 0xRustlang** [reply]
> > Some DNS servers do not work perfectly (even Google) In Iran. It's good for user to be free for choosing the best DNS server.
> 
> VpnHood changes DNS to 8.8.8.8 and tunnels the DNS request through VPN Tunnel, So it doesn't matter what you set for DNS because it will go through VPN tunnel. This issue exists somewhere else and can not be solved by changing the DNS.

Although I agree with you, I think ability to set the tunnel dns server have some benefits like using adguard DNS or other public DNS resolvers that have an ip address so the ads can get blocked inside tunnel (and not just browsers as user currently can use adblock DoH in them). 

BTW, I don't know about what benefits @AliMoal see when the DNS requests already goes through vpn tunnel.

**Reply by AliMoal** [reply]
Ok, the problem may be from somewhere else, I don't have any idea why this happens.
